cadquery/cq.py

```diff
@@ -2279,14 +2279,32 @@ def largestDimension(self):
         how long or wide a feature must be to make sure to cut through all of the material
         :return: A value representing the largest dimension of the first solid on the stack
         """
-        # TODO: this implementation is naive and returns the dims of the first solid... most of
-        # TODO: the time this works. but a stronger implementation would be to search all solids.
-        s = self.findSolid()
-        if s:
-            return s.BoundingBox().DiagonalLength * 5.0
-        else:
+        # Get all the solids contained within this CQ object
+        compound = self.findSolid()
+
+        # Protect against this being called on something like a blank workplane
+        if not compound:
             return -1
 
+        # Get the extents of the bounding box of the solids
+        xmin = compound.BoundingBox().xmin
+        ymin = compound.BoundingBox().ymin
+        zmin = compound.BoundingBox().zmin
+        xmax = compound.BoundingBox().xmax
+        ymax = compound.BoundingBox().ymax
+        zmax = compound.BoundingBox().zmax
+
+        # Find the length of each axis
+        xLength = xmax - xmin
+        yLength = ymax - ymin
+        zLength = zmax - zmin
+
+        # Calculate the sphere size of the outer bounds of all solids from the lengths along each axis
+        centroid = Vector(xLength, yLength, zLength)
+        sphereSize = centroid.Length
```

Comment:
I might be missing something but aren't you calculating the diagonal length by hand here? Wouldn't this work?
```
return compound.BoundingBox().DiagonalLength
```

Reply:
That's the way it was before, but it gives a slightly different answer that's smaller. My concern is if I use that the cut thru operations will start coming up short in some edge cases. However, I can use the built in diagonal length and then switch back to doing it by hand if users start encountering issues.

Reply:
Is it a large difference?

Reply:
It was between 0.5% and 1% in most of my testing, which wasn't terribly extensive.

Reply:
i think it is this line: https://github.com/CadQuery/cadquery/blob/b81319afd3a9586a61819da4666e9934fbdb1fac/cadquery/occ_impl/geom.py#L858

TOL=1e-2

Reply:
The PR has been updated to use DiagonalLength again. Essentially the only change is that the multiplication by 5 was removed. When that function was originally written, I don't think `findSolid` returned more than the first solid on the stack either, and that's since been fixed in #163